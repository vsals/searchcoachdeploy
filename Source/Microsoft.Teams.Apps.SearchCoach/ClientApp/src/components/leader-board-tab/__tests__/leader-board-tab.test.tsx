// <copyright file="leader-board-tab.test.tsx" company="Microsoft">
// Copyright (c) Microsoft. All rights reserved.
// </copyright>

import * as React from "react";
import { Provider } from "@fluentui/react-northstar";
import LeaderBoardTab from "../../leader-board-tab/leader-board-tab";
import { render } from "react-dom";
import { act } from "react-dom/test-utils";
import pretty from "pretty";

jest.mock("../../../api/leader-board-api");

jest.mock("react-i18next", () => ({
    useTranslation: () => ({
        t: (key: any) => key,
        i18n: { changeLanguage: jest.fn() },
    }),

    withTranslation: () => (Component: any) => {
        Component.defaultProps = {
            ...Component.defaultProps,
            t: (key: any) => key,
        };
        return Component;
    },
}));
jest.mock("@microsoft/teams-js", () => ({
    initialize: () => {
        return true;
    },
    getContext: (callback: any) =>
        callback(
            Promise.resolve({ teamId: "ewe", entityId: "sdsd", locale: "en-US" })
        ),
}));

let container: any = null;

beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement("div");
    // container *must* be attached to document so events work correctly.
    document.body.appendChild(container);
});

describe("LeaderBoardTab", () => {
    it("renders snapshots", () => {
        act(() => {
            render(
                <Provider>
                    <LeaderBoardTab />
                </Provider>,
                container
            );
        });

        expect(pretty(container.innerHTML)).toMatchSnapshot();
    });

    it("LeaderBoard Table Row Count Validation", () => {
        const leaderBoardDataTable = document.querySelector(
            "[data-testid=leader-board-table-data]"
        );
        expect(leaderBoardDataTable?.childElementCount).not.toBe(0);
    });
});
