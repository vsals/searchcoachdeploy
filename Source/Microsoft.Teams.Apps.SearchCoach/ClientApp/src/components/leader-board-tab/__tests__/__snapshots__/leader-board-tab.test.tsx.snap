// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderBoardTab renders snapshots 1`] = `
"<div class=\\"ui-provider\\" dir=\\"ltr\\" data-uses-unhanded-props=\\"true\\">
  <div>
    <div class=\\"ui-flex button-footer\\" data-uses-unhanded-props=\\"true\\"><button data-aa-class=\\"Button\\" class=\\"ui-button back-button\\" data-uses-unhanded-props=\\"true\\">
        <div class=\\"ui-box\\" data-uses-unhanded-props=\\"true\\"><span role=\\"img\\" aria-hidden=\\"true\\" class=\\"ui-icon\\"><svg role=\\"presentation\\" focusable=\\"false\\" viewBox=\\"8 8 16 16\\"><path d=\\"M9.1 15.9l4 4c.2.2.5.2.7 0s.2-.5 0-.7L10.7 16h11.8c.3 0 .5-.2.5-.5s-.2-.5-.5-.5H10.7l3.1-3.1c.2-.2.2-.5 0-.7 0-.2-.2-.2-.3-.2s-.3 0-.4.1l-4 4c-.1.2-.1.6 0 .8z\\"></path></svg></span></div><span dir=\\"auto\\" class=\\"ui-button__content\\" data-uses-unhanded-props=\\"true\\">back</span>
      </button><button data-aa-class=\\"Button\\" class=\\"ui-button\\" data-uses-unhanded-props=\\"true\\"><span dir=\\"auto\\" class=\\"ui-button__content\\" data-uses-unhanded-props=\\"true\\">Questions</span></button></div>
    <div class=\\"ui-flex content-items-body\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">points</span></div>
    <div class=\\"ui-flex content-items-body\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">pointsText</span></div>
    <div class=\\"ui-flex\\" data-uses-unhanded-props=\\"true\\">
      <div role=\\"table\\" data-aa-class=\\"Table\\" class=\\"ui-table user-data-table\\" data-testid=\\"leaderboard-table-data\\" data-uses-unhanded-props=\\"true\\">
        <div role=\\"row\\" data-aa-class=\\"TableRow\\" class=\\"ui-table__row ui-table__header\\" data-uses-unhanded-props=\\"true\\">
          <div role=\\"columnheader\\" data-aa-class=\\"TableCell\\" class=\\"ui-table__cell select-all-users\\" data-uses-unhanded-props=\\"true\\">
            <div class=\\"ui-box ui-table__cell__content\\" data-uses-unhanded-props=\\"true\\">
              <div aria-checked=\\"false\\" role=\\"checkbox\\" tabindex=\\"0\\" data-is-focusable=\\"true\\" data-aa-class=\\"Checkbox\\" class=\\"ui-checkbox\\" data-uses-unhanded-props=\\"true\\">
                <div class=\\"ui-box ui-checkbox__indicator\\" data-uses-unhanded-props=\\"true\\"></div>
              </div>
            </div>
          </div>
          <div role=\\"columnheader\\" data-aa-class=\\"TableCell\\" class=\\"ui-table__cell\\" data-uses-unhanded-props=\\"true\\">
            <div class=\\"ui-box ui-table__cell__content\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text user-name-header\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">name</span></div>
          </div>
          <div role=\\"columnheader\\" data-aa-class=\\"TableCell\\" class=\\"ui-table__cell questions-header\\" data-uses-unhanded-props=\\"true\\">
            <div class=\\"ui-box ui-table__cell__content\\" data-uses-unhanded-props=\\"true\\"><span class=\\"ui-text\\" dir=\\"auto\\" data-uses-unhanded-props=\\"true\\">questionsAnsweredRight</span></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>"
`;
